### PR TITLE
ui: Show ORT run and job statuses more clearly in the UI

### PR DESCRIPTION
**Pull request by Etsija**
This PR brings in some UI improvements for logging the statuses of ORT runs and their respective job statuses, concentrating mainly on the repository index page:
- all statuses are now polled periodically and the UI is updated accordingly
- runs are shown in reversed order of creation ie. newest runs are shown on top of the list/page
- run statuses are shown with coloured badges to more clearly indicate the general status of a run
- job statuses of a run are shown in the page with a "traffic lights" type of component

Please see the individual commits for details.

### CODE REVIEW

File: ui/src/components/ort-run-job-status.tsx
Comment (sschuberth) [inline ui/src/components/ort-run-job-status.tsx:1]: Commit message nit: Maybe "of the job" -> "about the job"?

File: ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx
Comment (sschuberth) [inline ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:64]: Just as I see it: This will probably trivially conflict with https://github.com/eclipse-apoapsis/ort-server/pull/233.

File: ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx
Comment (sschuberth) [inline ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:88]: Commit message title: "Revert" should say "Reverse".

Comment (sschuberth) [reply ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:88]: Same in the body text.

File: ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx
Comment (Etsija) [reply ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:64]: There will be conflicts, yes.

Comment (sschuberth) [reply ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:64]: We should simple rebase this once @mmurto's PR is merged and resolve the trivial conflicts along the way.

Comment (Etsija) [reply ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:63]: What do you mean with "along the way"? Should I make a separate commit and fix the quotes in every file this PR touches?  And what's the rule exactly: is it single quotes everywhere, so a search-and-place?

Comment (mmurto) [reply ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:63]: No need to search and replace just pnpm format.

Comment (mmurto) [reply ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:63]: As for handling conflicts, I'd say the easiest way is to first accept your changes and then run format and add that to the same commit.

Comment (Etsija) [reply ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:63]: I was actually talking about whether I need to do some formatting to this PR now - as I do not have `pnpm format`available until your PR is merged.

Comment (mmurto) [reply ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:63]: Probably makes sense to just wait for it, or you could rebase over it and wait for it to be merged. Though if we review the PR as-is very strictly, this change on lines 63/64 should probably not be part of the current commits, but a part of separate formatting commit, as the change of `"` to `'` is not related to these.

**Comment by sschuberth** [comment]
LGTM, but I prefer @mmurto or @lamppu to review as well.

File: ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx
Comment (mmurto) [inline ui/src/routes/_layout/organizations/$orgId.products.$productId.repositories.$repoId.index.tsx:73]: This should probably be outside of the component, so it doesn't try parsing the int on every render.

File: ui/src/components/ort-run-job-status.tsx
Comment (Etsija) [reply ui/src/components/ort-run-job-status.tsx:35]: Yeah, I tried to find the type of the `jobs` but couldn't at the time; I knew these props could be simplified.

File: ui/src/helpers/getStatusColors.ts
Comment (Etsija) [reply ui/src/helpers/getStatusColors.ts:40]: I need to use the helpers, otherwise Typescript complains about "undefined cannot be used as an index type".

Comment (Etsija) [reply ui/src/helpers/getStatusColors.ts:40]: Ie. I cannot access the const table directly, as in `STATUS_BACKGROUND_COLOR[status]`, because the status can be undefined.

**Review by mmurto — APPROVED** [review]

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**
